### ISSUE TITLE
Setting ServiceAccount name breaks permissions to cluster API

### ISSUE THREAD

**Issue by sharkeyl** [CloudHealth/helm#44]
Allowing users to set the ServiceAccount name [here](https://github.com/CloudHealth/helm/blob/main/cloudhealth-collector/values.yaml#L35) breaks the ClusterRoleBinding [here](https://github.com/CloudHealth/helm/blob/main/cloudhealth-collector/templates/clusterrolebinding.yaml#L15) due to the way the ServiceAccount name is handled in the `_helpers.tpl` file [here](https://github.com/CloudHealth/helm/blob/main/cloudhealth-collector/templates/_helpers.tpl#L59...L68). This results in the Collector not being able to open an input stream to cluster API, returning a 403 response. Please see below for error message for details:

```bash
D, [2022-04-26T15:26:17.197738 #13] DEBUG -- : loaded K8S config from  with master @ https://kubernetes.default.svc/ with ca certificate /var/run/secrets/kubernetes.io/serviceaccount/ca.crt with client_cert_file  with client key file  with trust_certs false with trust store file  with proxy username
D, [2022-04-26T15:26:19.194230 #13] DEBUG -- : Ensuring cache directory is present: /tmp/cache
D, [2022-04-26T15:26:19.488109 #13] DEBUG -- : Fetching state...
D, [2022-04-26T15:26:19.508177 #13] DEBUG -- : Connecting to URL: https://kubernetes.default.svc/api/v1/nodes
E, [2022-04-26T15:26:23.395960 #13] ERROR -- : [IOError]: Could not open an input stream to cluster API: Server returned HTTP response code: 403 for URL: https://kubernetes.default.svc/api/v1/nodes
E, [2022-04-26T15:26:23.405555 #13] ERROR -- : Backtrace:
	uri:classloader:/mesos-collector/lib/common/fetcher.rb:28:in `fetch'
	uri:classloader:/mesos-collector/bin/container-collector.rb:132:in `block in do_fetch_state'
	org/jruby/RubyHash.java:1415:in `each'
	uri:classloader:/mesos-collector/bin/container-collector.rb:131:in `do_fetch_state'
	uri:classloader:/mesos-collector/bin/container-collector.rb:114:in `block in do_upload'
	org/jruby/RubyKernel.java:1442:in `loop'
	uri:classloader:/mesos-collector/bin/container-collector.rb:109:in `do_upload'
	uri:classloader:/mesos-collector/bin/container-collector.rb:56:in `upload_k8s_state_v2'
	uri:classloader:/gems/boson-1.3.0/lib/boson/runner_library.rb:12:in `block in upload_k8s_state_v2'
	org/jruby/RubyMethod.java:131:in `call'
	uri:classloader:/gems/boson-1.3.0/lib/boson/scientist.rb:87:in `block in upload_k8s_state_v2'
	uri:classloader:/gems/boson-1.3.0/lib/boson/scientist.rb:123:in `call_original_command'
	uri:classloader:/gems/boson-1.3.0/lib/boson/scientist.rb:115:in `during_analyze'
	uri:classloader:/gems/boson-1.3.0/lib/boson/scientist.rb:107:in `analyze'
	uri:classloader:/gems/boson-1.3.0/lib/boson/scientist.rb:86:in `block in upload_k8s_state_v2'
	uri:classloader:/gems/boson-1.3.0/lib/boson.rb:79:in `full_invoke'
	uri:classloader:/gems/boson-1.3.0/lib/boson/bare_runner.rb:42:in `execute_command'
	uri:classloader:/gems/boson-1.3.0/lib/boson/runner.rb:35:in `execute_command'
	uri:classloader:/gems/boson-1.3.0/lib/boson/runner.rb:31:in `execute'
	uri:classloader:/gems/boson-1.3.0/lib/boson/runner.rb:26:in `start'
	uri:classloader:/mesos-collector/bin/container-collector.rb:198:in `<main>'
	org/jruby/RubyKernel.java:1009:in `load'
	uri:classloader:/META-INF/main.rb:1:in `<main>'
	org/jruby/RubyKernel.java:974:in `require'
	uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:54:in `require'
	<script>:3:in `<main>'
E, [2022-04-26T15:26:23.418309 #13] ERROR -- : Could not open an input stream to cluster API: Server returned HTTP response code: 403 for URL: https://kubernetes.default.svc/api/v1/nodes
E, [2022-04-26T15:26:23.487793 #13] ERROR -- : ATTENTION: The most common cause of this error is that the [CHT_API_TOKEN] or [CHT_CLUSTER_NAME] environment variables were not configured properly as part of agent deployment. Please check your full configuration, including the API token and the cluster name. After your config has been fixed, please wait 15 minutes for the changes to take effect, and collection to resume.
```

Either the ServiceAccount name should not be able to be set in the `values.yaml` file and instead overridden using something like `nameOverride` or `fullnameOverride`, or the ClusterRoleBinding should be updated to use the ServiceAccount name specified by the user.

**Comment by gm-cht** [comment]
Nice catch @sharkeyl . Let have a PR for this soon.

**Comment by gm-cht** [comment]
Fixed in Helm Chart version 1.1.2
